server/src/test/java/org/opensearch/index/remote/RemoteStoreStatsTrackerFactoryTests.java

```diff
@@ -0,0 +1,74 @@
+
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+        shardId = new ShardId("index", "uuid", 0);
+        indexShard = createIndexShard(shardId, true);
+        remoteStoreStatsTrackerFactory = new RemoteStoreStatsTrackerFactory(Settings.EMPTY);
+    }
+
+    public void testAfterIndexShardCreatedForRemoteBackedIndex() {
+        remoteStoreStatsTrackerFactory.afterIndexShardCreated(indexShard);
+        assertNotNull(remoteStoreStatsTrackerFactory.getRemoteSegmentTransferTracker(indexShard.shardId()));
+    }
+
+    public void testAfterIndexShardCreatedForNonRemoteBackedIndex() {
+        indexShard = createIndexShard(shardId, false);
+        remoteStoreStatsTrackerFactory.afterIndexShardCreated(indexShard);
+        assertNull(remoteStoreStatsTrackerFactory.getRemoteSegmentTransferTracker(indexShard.shardId()));
+    }
+
+    public void testAfterIndexShardClosed() {
+        remoteStoreStatsTrackerFactory.afterIndexShardCreated(indexShard);
+        assertNotNull(remoteStoreStatsTrackerFactory.getRemoteSegmentTransferTracker(shardId));
+        remoteStoreStatsTrackerFactory.afterIndexShardClosed(shardId, indexShard, indexShard.indexSettings().getSettings());
+        assertNull(remoteStoreStatsTrackerFactory.getRemoteSegmentTransferTracker(shardId));
+    }
+
+    public void testUpdateMovingAverageWindowSize() {
+        remoteStoreStatsTrackerFactory.afterIndexShardCreated(indexShard);
+
+        ShardId shardId2 = new ShardId("index", "uuid", 1);
+        IndexShard indexShard2 = createIndexShard(shardId2, true);
+        remoteStoreStatsTrackerFactory.afterIndexShardCreated(indexShard2);
+
+        int defaultSize = RemoteStorePressureSettings.Defaults.MOVING_AVERAGE_WINDOW_SIZE;
+        assertEquals(defaultSize, remoteStoreStatsTrackerFactory.getMovingAverageWindowSize());
+        assertFalse(remoteStoreStatsTrackerFactory.getRemoteSegmentTransferTracker(shardId).isUploadBytesAverageReady());
+        assertFalse(remoteStoreStatsTrackerFactory.getRemoteSegmentTransferTracker(shardId).isUploadBytesPerSecAverageReady());
+        assertFalse(remoteStoreStatsTrackerFactory.getRemoteSegmentTransferTracker(shardId).isUploadTimeMsAverageReady());
+        assertFalse(remoteStoreStatsTrackerFactory.getRemoteSegmentTransferTracker(shardId2).isUploadBytesAverageReady());
+        assertFalse(remoteStoreStatsTrackerFactory.getRemoteSegmentTransferTracker(shardId2).isUploadBytesPerSecAverageReady());
+        assertFalse(remoteStoreStatsTrackerFactory.getRemoteSegmentTransferTracker(shardId2).isUploadTimeMsAverageReady());
+
+        int updatedSize = 0;
+        assertThrows(
+            IllegalArgumentException.class,
+            () -> remoteStoreStatsTrackerFactory.updateMovingAverageWindowSize(
+                RemoteSegmentTransferTracker::updateMovingAverageWindowSize,
+                updatedSize
+            )
+        );
```

Comment:
`movingAverageWindowSize` field will get updated to the new value while the tracker update fails. While the setting has a validator of min value of 5, can we check if this behaviour can cause any issue?

Reply:
Updated `updateMovingAverageWindowSize` to change the field only after updating the trackers. I added an additional test to validate the min value. I am observing that a value less than min value but > 0 is successfully accepted. Need to revisit if we missed something.

Reply:
Updated. Resolving this thread. Please feel free to re-open if needed.